### PR TITLE
[ENG-6238][eas-cli] prompt developer to download the simulator app after build is finished when in interactive mode

### PR DESCRIPTION
**Pull request by szdziedzic**
# Checklist

- [ ] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

The last addition to the `eas build:run` feature

# How

If `eas build` is run in interactive mode and any of the builds is successfully finished simulator build prompt the user to install it on the emulator/simulator.

https://user-images.githubusercontent.com/55145344/205007745-62ca3f09-a671-436c-aadb-6393837e710a.mov

https://user-images.githubusercontent.com/55145344/205007773-c5d40293-cac4-47f6-8790-c217d7979fd5.mov

https://user-images.githubusercontent.com/55145344/205007792-f7b926d9-ed8f-4791-a583-ba5cc88eed25.mov

# Test Plan

Test manually.

### CODE REVIEW

File: packages/eas-cli/src/build/runBuildAndSubmit.ts [hunk 1]
```diff
@@ -242,6 +247,32 @@ export async function runBuildAndSubmitAsync(
...
+          message: `Would you like to automatically run the ${
+            simBuild.platform === AppPlatform.Android ? 'Android' : 'iOS'
+          } build on your ${simBuild.platform === AppPlatform.Android ? 'emulator' : 'simulator'}?`,
+        });
+        if (confirm) {
+          await downloadAndRunAsync(simBuild);
+        }
+      }
+    }
+  }
```
Comment (dsokal) [inline packages/eas-cli/src/build/runBuildAndSubmit.ts:274]: Move to a separate function, sth like `maybeDownloadAndRunSimulatorBuilds`

File: packages/eas-cli/src/build/runBuildAndSubmit.ts [hunk 2]
```diff
@@ -242,6 +247,32 @@ export async function runBuildAndSubmitAsync(
   const haveAllBuildsFailedOrCanceled = builds.every(
     build => build?.status && [BuildStatus.Errored, BuildStatus.Canceled].includes(build?.status)
   );
+
+  const simBuilds = builds.filter(
```
Comment (dsokal) [inline packages/eas-cli/src/build/runBuildAndSubmit.ts:251]:
```suggestion
  const simBuilds = builds.filter(truthy).filter(
```

so you don't need `build?` everywhere (+ `assert`)

File: packages/eas-cli/src/build/runBuildAndSubmit.ts [hunk 3]
```diff
@@ -242,6 +247,32 @@ export async function runBuildAndSubmitAsync(
...
+    build =>
+      build?.status === BuildStatus.Finished &&
+      !!build?.artifacts?.applicationArchiveUrl &&
+      (build?.distribution === DistributionType.Simulator ||
+        !build.artifacts.applicationArchiveUrl.endsWith('aab'))
```
Comment (dsokal) [inline packages/eas-cli/src/build/runBuildAndSubmit.ts:256]: I feel like those checks should be extracted to some functions. Aren't they used in other places? If so, it even makes more sense.

File: packages/eas-cli/src/build/runBuildAndSubmit.ts [hunk 4]
```diff
@@ -242,6 +247,32 @@ export async function runBuildAndSubmitAsync(
...
+  );
+
+  if (simBuilds.length > 0 && !flags.autoSubmit && !flags.nonInteractive) {
+    for (const simBuild of simBuilds) {
+      assert(simBuild?.platform);
+      if (simBuild.platform === AppPlatform.Android || process.platform === 'darwin') {
+        Log.newLine();
+        const confirm = await confirmAsync({
+          message: `Would you like to automatically run the ${
```
Comment (dsokal) [inline packages/eas-cli/src/build/runBuildAndSubmit.ts:265]:
```suggestion
          message: `Would you like to run the ${
```

File: packages/eas-cli/src/build/runBuildAndSubmit.ts [hunk 5]
```diff
@@ -242,6 +247,32 @@ export async function runBuildAndSubmitAsync(
...
+            simBuild.platform === AppPlatform.Android ? 'Android' : 'iOS'
+          } build on your ${simBuild.platform === AppPlatform.Android ? 'emulator' : 'simulator'}?`,
```
Comment (dsokal) [inline packages/eas-cli/src/build/runBuildAndSubmit.ts:267]:
```suggestion
          } build on ${simBuild.platform === AppPlatform.Android ? 'an emulator' : 'a simulator'}?`,
```

**Review by wkozyra95 — APPROVED** [review]

File: packages/eas-cli/src/build/runBuildAndSubmit.ts [hunk 6]
```diff
@@ -242,6 +247,32 @@ export async function runBuildAndSubmitAsync(
...
+
+  const simBuilds = builds.filter(
+    build =>
+      build?.status === BuildStatus.Finished &&
+      !!build?.artifacts?.applicationArchiveUrl &&
+      (build?.distribution === DistributionType.Simulator ||
+        !build.artifacts.applicationArchiveUrl.endsWith('aab'))
+  );
+
+  if (simBuilds.length > 0 && !flags.autoSubmit && !flags.nonInteractive) {
```
[the remark this answers is not in the document]
Comment (szdziedzic) [reply packages/eas-cli/src/build/runBuildAndSubmit.ts]: I think we were using it in my previous project this way - we ran sim builds with the `--platform all` flag and then opened them on simulators to make sure that everything works for sure for both platforms after the build. That's why I built it this way.

File: packages/eas-cli/src/run/utils.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+function isAab(build: BuildFragment): boolean {
+  return build.artifacts?.applicationArchiveUrl?.endsWith('.aab') ?? false;
+}
+
+function didArtifactsExpire(build: BuildFragment): boolean {
+  return new Date().getTime() - new Date(build.updatedAt).getTime() > 30 * 24 * 60 * 60 * 1000; // 30 days
+}
+
+export function isRunableOnSimulatorOrEmulator(build: BuildFragment): boolean {
```
Comment (dsokal) [inline packages/eas-cli/src/run/utils.ts:11]:
```suggestion
export function isRunnableOnSimulatorOrEmulator(build: BuildFragment): boolean {
```

**Review by dsokal — APPROVED** [review]

File: packages/eas-cli/src/build/runBuildAndSubmit.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (szdziedzic) [reply packages/eas-cli/src/build/runBuildAndSubmit.ts]: @brentvatne What do you think about it?

[the remark this answers is not in the document]
Comment (brentvatne) [reply packages/eas-cli/src/build/runBuildAndSubmit.ts]: I also don't have a strong opinion 😅 

I think prompting for both platforms on macOS seems OK if that is a use case you would have found useful @szdziedzic, and if people have some issues with that we can iterate on it.

File: packages/eas-cli/src/build/runBuildAndSubmit.ts [hunk 8]
```diff
@@ -409,3 +418,35 @@ function exitWithNonZeroCodeIfSomeBuildsFailed(maybeBuilds: (BuildFragment | nul
...
+  flags: BuildFlags
+): Promise<void> {
+  const simBuilds = builds.filter(truthy).filter(isRunnableOnSimulatorOrEmulator);
+
+  if (simBuilds.length > 0 && !flags.autoSubmit && !flags.nonInteractive) {
+    for (const simBuild of simBuilds) {
+      if (simBuild.platform === AppPlatform.Android || process.platform === 'darwin') {
+        Log.newLine();
+        const confirm = await confirmAsync({
+          message: `Would you like to run the ${
```
Comment (brentvatne) [inline packages/eas-cli/src/build/runBuildAndSubmit.ts:442]: what do you think about removing "Would you like to" here?

"Would you like to run the Android build on an emulator?"

->

"Install and run the Android build on an emulator?"

I think this is a bit more concise and to the point

**Comment by szdziedzic** [comment]
/changelog-entry new-feature Prompt developer to download the app after the simulator build is finished

**Pull request merged**